x/ecocredit/client/testsuite/query.go

```diff
@@ -485,3 +487,557 @@ func (s *IntegrationTestSuite) TestQueryParams() {
 
 	require.Equal(ecocredit.DefaultParams(), *params.Params)
 }
+
+func (s *IntegrationTestSuite) TestQuerySellOrder() {
+	val := s.network.Validators[0]
+	clientCtx := val.ClientCtx
+	clientCtx.OutputFormat = "JSON"
+	testCases := []struct {
+		name      string
+		args      []string
+		expErr    bool
+		expErrMsg string
+		expOrder  *ecocredit.SellOrder
+	}{
+		{
+			name:      "missing args",
+			args:      []string{},
+			expErr:    true,
+			expErrMsg: "Error: accepts 1 arg(s), received 0",
+		},
+		{
+			name:      "too many args",
+			args:      []string{"foo", "bar"},
+			expErr:    true,
+			expErrMsg: "Error: accepts 1 arg(s), received 2",
+		},
+		{
+			name:      "invalid sell order",
+			args:      []string{"foo"},
+			expErr:    true,
+			expErrMsg: "invalid sell order",
+		},
+		{
+			name:      "valid",
+			args:      []string{"1"},
+			expErr:    false,
+			expErrMsg: "",
+			expOrder: &ecocredit.SellOrder{
+				OrderId:           1,
+				Owner:             val.Address.String(),
+				BatchDenom:        batchDenom,
+				Quantity:          "1",
+				AskPrice:          &sdk.Coin{Denom: "regen", Amount: sdk.NewInt(100)},
+				DisableAutoRetire: false,
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		s.Run(tc.name, func() {
+			cmd := client.QuerySellOrderCmd()
+			out, err := cli.ExecTestCLICmd(clientCtx, cmd, tc.args)
+			if tc.expErr {
+				s.Require().Error(err)
+				s.Require().Contains(out.String(), tc.expErrMsg)
+			} else {
+				s.Require().NoError(err, out.String())
+
+				var res ecocredit.QuerySellOrderResponse
+				s.Require().NoError(clientCtx.Codec.UnmarshalJSON(out.Bytes(), &res))
+				s.Require().Equal(tc.expOrder, res.SellOrder)
+			}
+		})
+	}
+}
+
+func (s *IntegrationTestSuite) TestQuerySellOrders() {
+	val := s.network.Validators[0]
+	clientCtx := val.ClientCtx
+	clientCtx.OutputFormat = "JSON"
+	testCases := []struct {
+		name      string
+		args      []string
+		expErr    bool
+		expErrMsg string
+		expOrders []*ecocredit.SellOrder
+	}{
+		{
+			name:      "too many args",
+			args:      []string{"foo"},
+			expErr:    true,
+			expErrMsg: "Error: accepts 0 arg(s), received 1",
+		},
+		{
+			name:      "valid",
+			args:      []string{},
+			expErr:    false,
+			expErrMsg: "",
+			expOrders: []*ecocredit.SellOrder{
```

Comment:
could make these a top level constant so we dont have to copy pasta this sell order slice

Reply:
Added it to the test suite struct as it relies on val address https://github.com/regen-network/regen-ledger/pull/662/commits/7b0fabe694183b86062e2ed20f721922beca00a5